recon/src/protocol.rs

```diff
@@ -262,6 +264,7 @@ where
 
         self.role.close().await.context("closing stream")?;
         self.metrics.record(&ProtocolRun(start.elapsed()));
+        trace!("sync conversation complete: {}", self.role.sync_id());
```

Comment:
FYI that with tracing macros you can use `sync_id=%self.role.sync_id()` to get structured output with the field name in the logs.